db/system_keyspace.cc

```diff
@@ -1774,18 +1841,22 @@ std::vector<schema_ptr> all_tables() {
                     legacy::columns(), legacy::triggers(), legacy::usertypes(),
                     legacy::functions(), legacy::aggregates(), });
 
+    for (auto&& vt : virtual_tables) {
+        r.push_back(vt->schema());
+    }
+
     return r;
 }
 
-static void maybe_add_virtual_reader(schema_ptr s, database& db) {
-    if (s.get() == size_estimates().get()) {
-        db.find_column_family(s).set_virtual_reader(mutation_source(db::size_estimates::virtual_reader(db)));
-    }
-    if (s.get() == v3::views_builds_in_progress().get()) {
-        db.find_column_family(s).set_virtual_reader(mutation_source(db::view::build_progress_virtual_reader(db)));
-    }
-    if (s.get() == built_indexes().get()) {
-        db.find_column_family(s).set_virtual_reader(mutation_source(db::index::built_indexes_virtual_reader(db)));
+static void install_virtual_readers(database& db) {
```

Comment:
I think you squashed some refactoring into this commit, which could have stand on its own for clarity. But I guess we can let this slip.